### PR TITLE
ci: remove sharp for developers

### PR DESCRIPTION
**Pull request by Enter-tainer**

### CODE REVIEW

**Comment by Enter-tainer** [comment]
ready for review

File: package.json [hunk 1]
```diff
@@ -4,13 +4,14 @@
   "version": "0.0.0",
   "license": "MIT",
   "scripts": {
+    "precibuild": "npm i gatsby-plugin-sharp gatsby-transformer-sharp gatsby-remark-images gatsby-plugin-manifest && sed -i 's/\\/\\/ CI BUILD //g' gatsby-config.js",
```
Comment (diauweb) [inline package.json:7]: i'm afraid these packages can no longer being version controlled by npm and the build could be broken due to any breaking changes in these packages.
it does not promise builds built in different times to be working consistently, and may fail silently **in build** without developing environment notice

File: gatsby-config.js [hunk 2]
```diff
@@ -16,22 +16,23 @@ module.exports = {
         path: './docs/',
       },
     },
-    'gatsby-plugin-sharp',
-    'gatsby-transformer-sharp',
+    // 以 CI BUILD 开头的注释会在构建时被取消注释。
```
Comment (diauweb) [inline gatsby-config.js:19]: i have to say this is a javascript file and we can control its behavior by a less 'dirty' way, like reading environment variables or something else. seding this file seems too dirty for me

Comment (Enter-tainer) [reply gatsby-config.js:19]: 可以做两份配置，根据配置文件启用？

Comment (diauweb) [reply gatsby-config.js:19]: i expect something like
```js
/* foo bar */ },
isProd(() => ({
   resolve: 'random-plugin',
   ...
})),
{ /* bar baz*/ }, ...
```

Comment (Enter-tainer) [reply gatsby-config.js:19]: 那就在 ci 配置文件里面写一下环境变量，然后在 gatsby-config.js 里面根据配置文件决定是否启用吧

File: package.json (hunk 1 above)
Comment (Enter-tainer) [reply package.json:7]: 这个确实是没办法的事情，考虑以后这个变成主站，让每一个人都能顺利的编译&&下载好 sharp 是不太现实的，至少在本项目的 contrib 中，还有不少人非常受这个东西困扰。

不知道有没有更好的解决方案

Comment (diauweb) [reply package.json:7]: `optionalDependencies` may help
and who may have problems in internet connection can simply `npm install --no-optional`

Comment (Enter-tainer) [reply package.json:7]: lgtm

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -31,4 +31,6 @@ jobs:
 
       - run: npm install
 
-      - run: npm run cibuild
+      - run: |
+          export PRODUCTION=true
```
Comment (diauweb) [inline .github/workflows/build-test.yml:35]: i don't know but does this works on Windows?

File: gatsby-config.js [hunk 4]
```diff
@@ -1,5 +1,9 @@
 const path = require('path')
-
+const isProd = process.env.PRODUCTION === 'true'
+const condition = (cond, v) => cond ? [v] : []
+if (isProd && process.env.gatsby_executing_command !== 'build') {
+  console.warn('You are in procution mode')
```
Comment (diauweb) [inline gatsby-config.js:5]:
```suggestion
  console.warn('Using production configurations in non-build environment')
```

**Comment by Enter-tainer** [comment]
render 的以后再搞吧。

只需要加一个环境变量就可以了，不需要动 build 的命令

**Pull request merged**